zeppelin-zengine/pom.xml

```diff
@@ -657,6 +625,69 @@
             </exclusion>
           </exclusions>
         </dependency>
+
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-azure</artifactId>
+          <version>${hadoop.version}</version>
+          <exclusions>
+            <exclusion>
+              <groupId>com.fasterxml.jackson.core</groupId>
+              <artifactId>jackson-core</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>com.google.guava</groupId>
+              <artifactId>guava</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>org.apache.commons</groupId>
+              <artifactId>commons-lang3</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>com.jcraf</groupId>
+              <artifactId>jsch</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>org.apache.commons</groupId>
+              <artifactId>commons-compress</artifactId>
+            </exclusion>
+          </exclusions>
+        </dependency>
+        <dependency>
+          <groupId>com.microsoft.azure</groupId>
+          <artifactId>azure-data-lake-store-sdk</artifactId>
+          <version>${adl.sdk.version}</version>
+          <exclusions>
+            <exclusion>
+              <groupId>com.fasterxml.jackson.core</groupId>
+              <artifactId>jackson-core</artifactId>
+            </exclusion>
+          </exclusions>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-aws</artifactId>
```

Comment:
shouldn't this be under some vendor profile instead of building with all of azure + aws all the time?

Reply:
hmm, make sense

Reply:
Think about it twice. It might not be proper to put in vender profile. Because zeppelin's builtin AzureNotebookRepo and S3NotebookRepo has some dependencies which is dependency of hadoop-azure and hadoop-aws.

Reply:
true... I think they should have a plugin model then, like interpreter, can be installed as needed.

Reply:
That's correct. I have a plan to introduce plugin framework to zeppelin as zeppelin provide more extension point now like conf storage, notebook storage, lifecycle manager, interpreter launcher and etc. Plugin framework can help manage these plugins and create isolated classpath loader for them so that we don't need to worry about the jar conflict between them. But it would be in the next major release at least.